### PR TITLE
Adding MeanAbsoluteError, MeanAbsolutePercentageError, MeanSquaredLogarithmicError and BinaryCrossentropy loss classes.

### PR DESCRIPTION
**Pull request by pavithrasv**
### Summary
Adding MeanAbsoluteError, MeanAbsolutePercentageError, MeanSquaredLogarithmicError and BinaryCrossentropy loss classes.

### Related Issues
n/a

### PR Overview

- [ y] This PR requires new unit tests [y/n] (make sure tests are included)
- [ n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y ] This PR is backwards compatible [y/n]
- [ n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/losses_test.py [hunk 1]
```diff
@@ -201,5 +201,349 @@ def test_sum_reduction(self):
...
+        # (100 - 100 * (1 - 0.5 L)  + 0 +
+        #  0   + 100 * (0.5 L)      + 0 +
+        #  0   + 100 * (1 - 0.5 L)  + 0) * (1/3)
+        #  = (100 + 50L) * 1/3
+        bce_obj = losses.BinaryCrossentropy(
+            from_logits=True, label_smoothing=label_smoothing)
+        loss = bce_obj(y_true, logits)
+        expected_value = (100.0 + 50.0 * label_smoothing) / 3.0
+        print('LOSS VALUE', K.eval(loss))
+        print('EXPECTED VALUE', expected_value)
```
Comment (fchollet) [inline tests/keras/losses_test.py]: Print statements not necessary.

Comment (pavithrasv) [reply tests/keras/losses_test.py]: Done. Forgot to remove them.
